kytos/core/auth.py

```diff
@@ -44,6 +50,83 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class UserController:
+    """UserController"""
+
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, get_mongo=lambda: Mongo()):
+        self.mongo = get_mongo()
+        self.db_client = self.mongo.client
+        self.db = self.db_client[self.mongo.db_name]
+
+    def bootstrap_indexes(self):
+        """Bootstrap all topology related indexes."""
+        index_tuples = [
+            ("users", [("username", pymongo.ASCENDING)]),
+        ]
+        for collection, keys in index_tuples:
+            if self.mongo.bootstrap_index(collection, keys):
+                LOG.info(
+                    f"Created DB index {keys}, collection: {collection}"
+                )
+
+    def create_user(self, user_data):
+        """Create user to database"""
+        utc_now = datetime.datetime.utcnow()
+        try:
+            data = {
+                "username": user_data["username"],
+                "email": user_data["email"],
+                "password": user_data["password"],
+                "inserted_at": utc_now,
+                "updated_at": utc_now,
+            }
+            user = UserDoc(**data)
+            result = self.db.users.insert_one(user.dict())
+        except (ValueError, KeyError, Exception) as e:
+            raise e
+        return result
+
+    def delete_user(self, username):
+        """Delete user from database"""
+        utc_now = datetime.datetime.utcnow()
+        update = {
+                "state": "inactive",
+                "deleted_at": utc_now,
+        }
+        result = self.db.users.find_one_and_update(
+            {"username": username},
+            {"$set": update},
+            return_document=ReturnDocument.AFTER,
+        )
+        return result
+
+    def update_user(self, username, data):
+        """Update user from database"""
+        data.update({"updated_at": datetime.datetime.utcnow()})
+        result = self.db.users.find_one_and_update(
+            {"username": username},
+            {"$set": data},
+            return_document=ReturnDocument.AFTER
+        )
+        return result
+
+    def get_user(self, username):
+        """Return a user information from database"""
+        data = self.db.users.aggregate([
+            {"$match": {"username": username}},
+            {"$project": UserDoc.projection()}
+        ])
+        return {"data": value for value in data}
```

Comment:
Aldo, no need to try to keep the old `data` dict key that store house used to impose, let's get rid of the `data` everywhere else too. All docstrings on other methods that mention storehouse should also be updated.

Reply:
Ok. It will directly show the data:
```
{
    "deleted_at": "2022-10-19T03:08:07",
    "email": "alopalao@hotmail.com",
    "inserted_at": "2022-10-19T03:07:31",
    "updated_at": "2022-10-19T03:07:31",
    "username": "alopalao2"
}
```

Reply:
Sounds good. We also need to include `state` as well. Although you could tell that if `deleted_at` is set then it's inactive, but we need `state` there for explicitness, in the future we'll likely also have other states here, like email not confirmed or any other ones that are aligned with the use cases the we'll have (but that's far away, mentioning here for clarity).

Reply:
Added.